proseco/acq.py

```diff
@@ -126,6 +128,23 @@ class AcqTable(ACACatalogTable):
     _fid_set = MetaAttribute(is_kwarg=False, default=())
     imposters_mag_limit = MetaAttribute(is_kwarg=False, default=20.0)
 
+    def __setstate__(self, state):
+        """Set self during unpickling.
+        This has special handling to deal with restoring the ``acqs`` weak
+        reference in the AcqProbs objects.  Since weakrefs cannot be pickled,
+        they are simply dropped prior to pickling and restored here.
+        """
+        super().__setstate__(state)
+
+        # This could be a cand_acqs table or acqs table, so check if
+        # ``cand_acqs`` has something, and if so then create the weakref in
+        # each of the AcqProbs objects stored in the ``probs`` column.  TO DO:
+        # make two separate classes AcqTable and CandAcqTable to avoid this
+        # contextual hack.
+        if self.cand_acqs is not None:
+            for probs in self.cand_acqs['probs']:
+                probs.acqs = weakref.ref(self)
```

Comment:
This looks reasonable to me, but I'm not as clear about how we get to this point in the code.  With that in mind does this need any new tests besides the two assert statements added in test_catalog?

Reply:
This is basically the unpickling post-processing code, so gets called when an AcqTable is unpickled.  The existing tests (even without the two new asserts) will give an exception without this code.  I've added a comment in the test code to explain why the asserts there are "strong" tests of this new functionality, agreed it wasn't obvious just from the asserts.